### PR TITLE
fix `Should emulate a native browser behaviour for non success status codes of fetch responces` (close #1397)

### LINKED ISSUE #1397: Should emulate a native browser behaviour for non success status codes of fetch responces
see https://github.com/DevExpress/testcafe/issues/1973

Example for repoduce:

#### index.js
```js
var http = require('http');
var fs   = require('fs');

http.createServer(function (req, res) {
    var content = '';

    if (req.url === '/')
        content = fs.readFileSync('index.html');
    else if (req.url === '/500') {
        res.statusCode    = 500;
        res.statusMessage = 'error message';
    }

    res.end(content);
}).listen(3000);
```
#### index.html
```html
<html>
	<body>
		<script>		
			fetch('/500')
				.then(function (res) {
					console.log(res.status);
					console.log(res.statusText);
                });
		</script>
	</body>
</html>
html
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @AndreyBelym

### CODE REVIEW

File: test/server/proxy-test.js [hunk 1]
```diff
@@ -806,6 +806,21 @@ describe('Proxy', () => {
...
+        it('Should generate an error if an error occurred when attempting to contact with destination server', done => {
+            const options = {
+                url:     proxy.openSession('https://127.0.0.1:2000/', session),
+                headers: {
+                    [XHR_HEADERS.fetchRequestCredentials]: 'omit'
+                }
+            };
+
+            request(options, err => {
+                expect(err).not.to.be.null;
```
Comment (AndreyBelym) [inline test/server/proxy-test.js]: Maybe some more specific assertion, e.g. for error message, here?

File: test/server/proxy-test.js [hunk 2]
```diff
@@ -1694,10 +1709,10 @@ describe('Proxy', () => {
 
             const requestTime = Date.now();
 
-            request(options, (err, res) => {
+            request(options, err => {
                 const responseTime = Date.now();
 
-                expect(res.statusCode).eql(500);
+                expect(err).not.to.be.null;
```
Comment (AndreyBelym) [inline test/server/proxy-test.js]: And there

File: test/client/config-qunit-server-app.js [hunk 3]
```diff
@@ -113,6 +113,10 @@ module.exports = function (app) {
         res.send('Server error');
     });
 
+    app.get('/generate-client-net-error', function (req) {
```
Comment (miherlosev) [inline test/client/config-qunit-server-app.js]: `app.get('/close-request',...`

File: test/client/fixtures/sandbox/fetch-test.js [hunk 4]
```diff
@@ -332,23 +333,51 @@ if (window.fetch) {
...
-                .then(function (response) {
-                    return response.text();
-                })
-                .then(function () {
-                    ok(false, 'request promise should be rejected');
-                })
-                .catch(function (err) {
-                    ok(err instanceof TypeError);
-                });
+        test('request promise should emulate native behavior with error and error status code (GH-1397)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/fetch-test.js]: `should emulate a native browser behavior for fetch request with non success status code`

File: test/client/fixtures/sandbox/fetch-test.js [hunk 5]
```diff
@@ -332,23 +333,51 @@ if (window.fetch) {
...
+            var makeFetch = function (fetchFn, url) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/fetch-test.js]: `var performRequest` here and above

File: test/client/fixtures/sandbox/xhr-test.js [hunk 6]
```diff
@@ -239,3 +241,49 @@ asyncTest('"XHR_COMPLETED_EVENT" should be raised when xhr is prevented (GH-1283
     xhr.open('GET', '/xhr-test/', true);
     xhr.send();
 });
+
+test('xhr request should emulate native behavior with error and error status code (GH-1397)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/xhr-test.js]: should emulate a native browser behavior for xhr request with non success status code

File: test/server/proxy-test.js [hunk 7]
```diff
@@ -806,6 +806,21 @@ describe('Proxy', () => {
                 });
             });
         });
+
+        it('Should generate an error if an error occurred when attempting to contact with destination server', done => {
```
Comment (miherlosev) [inline test/server/proxy-test.js]: `Should respond with error if an error is occurred on attempting to connect with destination server`

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**